shared/libraries/config_protocol/include/config_protocol/config_client_property_object_impl.h

```diff
@@ -631,6 +631,8 @@ void ConfigClientPropertyObjectBaseImpl<Impl>::cloneAndSetChildPropertyObject(co
                                      });
         
         const auto impl = dynamic_cast<ConfigClientPropertyObjectImpl*>(clientPropObj.getObject());
+        if (impl == nullptr)
+            throw InvalidStateException("Failed to cast to ConfigClientPropertyObjectImpl");
```

Comment:
This feels like it's masking an underlying issue. This should be changed to an assert, as the dynamic_cast line should never be reached, if we are not working with a `ConfigClientPropertyObjectImpl` object.